hpy/devel/src/runtime/ctx_tuplebuilder.c

```diff
@@ -0,0 +1,53 @@
+#include <stddef.h>
+#include <Python.h>
+#include "hpy.h"
+
+#ifdef HPY_UNIVERSAL_ABI
+   // for _h2py and _py2h
+#  include "handles.h"
+#endif
+
+
+_HPy_HIDDEN HPyTupleBuilder
+ctx_TupleBuilder_New(HPyContext ctx, HPy_ssize_t initial_size)
+{
+    PyObject *tup = PyTuple_New(initial_size);
+    if (tup == NULL)
+        PyErr_Clear();   /* delay the MemoryError */
```

Comment:
why do you want to delay it?

Reply:
It's like in ListBuilder.  I'll write a comment.